src/serial.h

```diff
@@ -0,0 +1,30 @@
+#ifndef SERIAL_H
+#define SERIAL_H
+
+#include <pthread.h>
+#include <stdint.h>
+
+#define COM1_PORT_BASE 0x03f8
+#define COM1_PORT_SIZE 8
+#define COM1_PORT_END (COM1_PORT_BASE + COM1_PORT_SIZE)
+
+#define FIFO_LEN 64
+
+typedef struct serial_dev serial_dev_t;
+
+struct serial_dev {
+    void *priv;
+    pthread_mutex_t mutex;
```

Comment:
Rename from `mutex` to `lock` because we might re-implement with something like coroutines later.